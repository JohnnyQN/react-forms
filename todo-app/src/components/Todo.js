import React from "react";

function Todo({ id, task, removeTodo }) {
  return (
    <li>
      {task}
      <button onClick={() => removeTodo(id)}>X</button>
    </li>
  );
}

export default Todo;
