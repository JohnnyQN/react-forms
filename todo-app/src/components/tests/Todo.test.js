import { render, fireEvent } from "@testing-library/react";
import Todo from "../Todo";

// Smoke test
test("renders without crashing", () => {
  render(<Todo id={1} task="Test Todo" removeTodo={() => {}} />);
});

// Snapshot test
test("matches snapshot", () => {
  const { asFragment } = render(<Todo id={1} task="Test Todo" removeTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

// Test remove functionality
test("calls removeTodo function on button click", () => {
  const removeTodoMock = jest.fn();
  const { getByText } = render(<Todo id={1} task="Test Todo" removeTodo={removeTodoMock} />);
  
  fireEvent.click(getByText("X"));
  expect(removeTodoMock).toHaveBeenCalledWith(1);
});
