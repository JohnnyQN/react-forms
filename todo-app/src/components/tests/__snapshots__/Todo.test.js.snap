// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    Test Todo
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
